spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSQLContext.scala

```diff
@@ -36,199 +31,48 @@ import collection.mutable
   *
   * }}} */
 class CassandraSQLContext(sc: SparkContext) extends SQLContext(sc) {
```

Comment:
This is the second thing called Cassandra SQL context and now I'm quite confused. The other one is in `org.apache.spark.sql.cassandra` package, in a form of an implicit wrapper over Spark SQLContext.

```scala
 /**
   * Add a method, `cassandraTable`, to SQLContext that allows reading from and writing to Cassandra.
   * Add methods to access cluster level, keyspace level and table level Cassandra configuration settings.
   */
  implicit class CSQLContext(sqlContext: SQLContext) {
```
Some similarities:
- both need an SQLContext to work (one uses inheritance, another one composition)
- both offer methods to access Cassandra data as `DataFrame`
- both have methods to change config / e.g. set some table-level options; 
- they look like an "entry point" of the API
- and they have almost the same name `CSQLContext` and `CassandraSqlContext`

However, they offer different APIs, and one persists its state in memory, while another in Cassandra (in the metastore). 

As a user of this, which class should I use? Why are they two different "things"? What is the main purpose of each of them? I can see no clear separation in their functionality.